skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayer.kt

```diff
@@ -73,7 +84,19 @@ open class SkiaLayer() : HardwareLayer() {
             }
 
             renderer?.onRender(canvas!!, width, height)
-            context!!.flush()
+
+            if (api == GraphicsApi.RASTER) {
+                var img: BufferedImage?
+                val bais = ByteArrayInputStream(surface!!.makeImageSnapshot().encodeToData()!!.bytes)
```

Comment:
Looks like very inefficient solution, cannot we instead get NIO byte buffer wrapping image content instead?